hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java

```diff
@@ -239,7 +242,7 @@ public int getKey() {
             });
             return true;
         } finally {
-            plane.connectionsInProgress.remove(remoteAddress);
+            unblock(plane, remoteAddress);
```

Comment:
This call can now block this thread, which is inside a synchronized method. It seems quite bad and I'm not sure if this is the right approach. Same in `reset` and other methods invoking `unblock`.

Reply:
this is part of a condition signal() which requires a lock. This is not a "real" lock because it locks only for condition.signal()

Reply:
no longer using signal / loci